tensorflow_probability/python/mcmc/replica_exchange_mc.py

```diff
@@ -969,46 +962,19 @@ def _swap_then_retemper(x):
       x = x[0]
     return x
 
-  if isinstance(kr.accepted_results,
-                hmc.UncalibratedHamiltonianMonteCarloKernelResults):
-    kr = kr._replace(
-        accepted_results=kr.accepted_results._replace(
-            target_log_prob=_swap_then_retemper(
-                kr.accepted_results.target_log_prob),
-            grads_target_log_prob=_swap_then_retemper(
-                kr.accepted_results.grads_target_log_prob)))
-  elif isinstance(kr.accepted_results,
-                  random_walk_metropolis.UncalibratedRandomWalkResults):
-    kr = kr._replace(
-        accepted_results=kr.accepted_results._replace(
-            target_log_prob=_swap_then_retemper(
-                kr.accepted_results.target_log_prob)))
-  else:
-    # TODO(b/143702650) Handle other kernels.
-    raise NotImplementedError(
-        'Only HMC and RWMH Kernels are handled at this time. Please file a '
-        'request with the TensorFlow Probability team.')
+  kr = _update_field(kr, 'target_log_prob', _swap_then_retemper(
+      _get_field(kr, 'target_log_prob')))
+  try:
+    new_grads_target_log_prob = _swap_then_retemper(
+        _get_field(kr, 'grads_target_log_prob'))
+    kr = _update_field(
+        kr, 'grads_target_log_prob', new_grads_target_log_prob)
+  except NotImplementedError:
+    pass
```

Comment:
This merits at least a comment saying where you expect that `NotImplementedError` to come from.

Reply:
I added a detailed comment. But
> This merits **at least** a comment .... (emphasis mine)

sounds like this might still not be optimal?

Reply:
Well, there's optimal, and there's worth the time to perfect.  This looks good to me as it is.